x-pack/plugins/security_solution/common/endpoint/types/actions.ts

```diff
@@ -271,6 +271,11 @@ export interface ActionDetails<TOutputContent extends object = object> {
    * This is an Array because the action could have been sent to multiple endpoints.
    */
   agents: string[];
+  /**
+   * The endpoint ID/fleet agent ID which was used to create the action (same as above) and the agent's name
+   * Action could have been sent for multiple endpoints which is why this is an Array
+   */
+  hosts: Array<{ id: string; name: string }>;
```

Comment:
I think `name` needs to be `optional` here. You are not guaranteed that  the agent id still exists in the system.

Reply:
we're changing it to a `Record<>` right?

So it would be:

```typescript
hosts?: {
  [agentId: string]: string | undefined
}
```

or shorthand:

```typescript
hosts?: Record<string, undefined | string>
```

Reply:
👍

Reply:
https://github.com/elastic/kibana/pull/139379/commits/101408939af5c38e5e2045e8afbfe99b544f9f53#diff-f7586e6a6b97a380fd6848d32994e7b5e60a6bf1c579bb6b6efeea2df455597dL278-R278